fah_alchemy/base/api.py

```diff
@@ -148,6 +148,22 @@ def scope_params(org: str = None, campaign: str = None, project: str = None):
         )
 
 
+def _check_store_connectivity(n4js: Neo4jStore, s3os: S3ObjectStore) -> dict:
+    """Check if neo4j and s3 object store are reachable"""
+    # check if neo4j database is reachable
+    neo4jreachable = n4js._store_check()
+    # check if s3 object store is reachable
+    s3reachable = s3os._store_check()
+
+    if not neo4jreachable or not s3reachable:
+        detail = f"Attempt to reach services failed, Neo4j reachable: {neo4jreachable}, S3 reachable: {s3reachable}"
+        raise HTTPException(
+            status_code=status.HTTP_503_SERVICE_UNAVAILABLE, detail=detail
+        )
+    else:
+        return status.HTTP_200_OK
```

Comment:
We don't need to return a status code, and this isn't how `fastapi` handles them. I'll return `True` here instead just to indicate everything's good if no exception is raised.

Reply:
The fastapi docs detail how to make an API endpoint give a different status code on success: https://fastapi.tiangolo.com/tutorial/response-status-code/ ; otherwise for failure the exception we raise here will apply.